tasks/set_vars.yml

```diff
@@ -5,6 +5,12 @@
   when: not ansible_facts.keys() | list |
     intersect(__postgresql_required_facts) == __postgresql_required_facts
 
+- name: Add memory information used by the role
+  setup:
+    filter: ansible_memory_mb
+  when: not ansible_facts.keys() | list |
+    intersect(__postgresql_required_facts) == __postgresql_required_facts
```

Comment:
you don't need to add a separate fact gathering task only for the `ansible_memory_mb`

Reply:
ok, the role needs memory information so another solution could be to gather all hardware facts instead of a task with a filter. Is it good practice?

Reply:
> ok, the role needs memory information so another solution could be to gather all hardware facts instead of a task with a filter. Is it good practice?

https://github.com/linux-system-roles/postgresql/blob/0e24ed6fb9e6fe0e43270bcbc6d8893ffdde6a1d/tasks/set_vars.yml#L2
Adding `ansible_memory_mb` to `__postgresql_required_facts` should be enough - however, you may need to change `gather_subset: min` to some other subset that includes `ansible_memory_mb` if `min` does not include memory - https://github.com/linux-system-roles/postgresql/blob/0e24ed6fb9e6fe0e43270bcbc6d8893ffdde6a1d/tasks/set_vars.yml#L4 
https://docs.ansible.com/ansible/2.9/modules/setup_module.html#parameter-gather_subset - maybe `hardware` will include `ansible_memory_mb`?

The problem is that gathering facts is an expensive operation - better to do it only once rather than doing it twice.

Reply:
Yes, that's it, min subset doesn't include memory information, but the hardware does. My question was whether adding a hardware subset is good practice. So thanks for answer I will change it.